request_test.go

```diff
@@ -256,6 +256,104 @@ func TestAsyncRequestWithoutContext(t *testing.T) {
 	}
 }
 
+func TestAsyncRequestWithoutContextFailure(t *testing.T) {
+
+	ts := newServer(response{200, `
+			{
+				"asyncjobresult": {
+					"jobid": "1",
+					"jobresult": {},
+					"jobstatus": 0
+				}
+			}
+			`}, response{200, `
+			{"queryasyncjobresultresponse": {
+					"jobid": "1",
+					"jobresult": {
+						"virtualmachine": []
+					},
+					"jobstatus": 1
+			}}`}, response{200, `
+				{
+					"asyncjobresult": {
+						"jobid": "1",
+						"jobresult": {},
+						"jobstatus": 0
+					}
+				}
+				`}, response{200, `
+				{"queryasyncjobresultresponse": {
+					"jobid": "1",
+					"jobresult": {
+						"virtualmachine": {}
+					},
+					"jobstatus": 2
+				}}`}, response{200, `
+				{
+					"asyncjobresult": {
+					"jobid": "1",
+					"jobresult": {},
+					"jobstatus": 0
+				}
+				}
+				`}, response{200, `
+					{"queryasyncjobresultresponse": {
+						"jobid": "1",
+						"jobresult": [],
+						"jobstatus": 1
+				}}`})
+
+	defer ts.Close()
+
+	cs := NewClient(ts.URL, "TOKEN", "SECRET")
+	req := &DeployVirtualMachine{
+		Name:              "test",
+		ServiceOfferingID: "71004023-bb72-4a97-b1e9-bc66dfce9470",
+		ZoneID:            "1128bd56-b4d9-4ac6-a7b9-c715b187ce11",
+		TemplateID:        "78c2cbe6-8e11-4722-b01f-bf06f4e28108"}
+
+	resp := &DeployVirtualMachineResponse{}
+
+	// WithContext
+	cs.AsyncRequest(req, func(j *AsyncJobResult, err error) bool {
+		if err != nil {
+			t.Error(err)
+		}
+
+		if j.JobStatus == Success {
+
+			if r := j.Response(resp); r != nil {
+				return false
+			}
+			t.Errorf("Expected an error, got <nil>")
+		}
+		return true
+	})
+
+	cs.AsyncRequest(req, func(j *AsyncJobResult, err error) bool {
+		if err != nil {
+			return false
+		}
+		return true
+	})
+
+	cs.AsyncRequest(req, func(j *AsyncJobResult, err error) bool {
+		if err != nil {
+			t.Error(err)
```

Comment:
`Fatal`

Reply:
Put Fatal instead of t.Error() ?

Reply:
![](http://lecoinzic.info/wp-content/uploads/img/Fatal+Bazooka1.jpg)